shopBuddy/shopBuddy/ShoppingListManagerViewController.m

```diff
@@ -0,0 +1,69 @@
+//
+//  ShoppingListManagerViewController.m
+//  shopBuddy
+//
+//  Created by Rachna Gupta on 7/18/22.
+//
+
+#import "ShoppingListManagerViewController.h"
+#import "ShoppingListCell.h"
+#import "ShoppingList.h"
+#import "SpecificListViewController.h"
+#import "ShoppingList+Persistent.h"
+@interface ShoppingListManagerViewController () <UITableViewDataSource, UITableViewDelegate>
+{
+    __weak IBOutlet UITableView *tableView;
+    NSArray<ShoppingList*> *lists;
+    
+}
+
+@end
+
+@implementation ShoppingListManagerViewController
+
+- (void)viewDidLoad {
+    tableView.dataSource = self;
+    tableView.delegate = self;
+    tableView.rowHeight = UITableViewAutomaticDimension;
+    [ShoppingList fetchListsByUser:[PFUser currentUser] withCompletion:^(NSArray *lists, NSError *error) {
+        self->lists = lists;
+        [self->tableView reloadData];
+    }];
+    [super viewDidLoad];
+}
+
+- (IBAction)didTapAddList:(id)sender {
+    //TODO: add List names (make List of stores)
+    [ShoppingList createEmptyList: @"New_List" withCompletion:^(BOOL succeeded, NSError *error) {}];
+    [ShoppingList fetchListsByUser:[PFUser currentUser] withCompletion:^(NSArray *lists, NSError *error) {
+        self->lists = lists;
+        [self->tableView reloadData];
+    }];
```

Comment:
These two are independent asynchronous calls, you are creating a race condition here since `createEmptyList ` is not guaranteed to be finished before you send `fetchListsByUser` therefore the list might not exist if the second call gets through first.

You need to sequence both calls by basically calling `fetchListsByUser` inside the completion block of `createEmptyList`.